src/ct_eval.cpp

```diff
@@ -107,6 +107,19 @@ TypedAST::TypedAST* ct_eval(
 	return ast;
 }
 
+TypedAST::TypedAST* ct_eval(
+    TypedAST::MatchExpression* ast, TypeChecker& tc, TypedAST::Allocator& alloc) {
+	// NOTE: no need to ct_eval the identifier, because it is guaranteed
+	// to be of metatype value, thus nothing needs to be done
```

Comment:
Since we should allow expressions in there (which can have type expressions within them), we should `ct_eval` the matchee